diff --git a/imgui.h b/imgui.h
index c7f53448..1288f9a5 100644
--- a/imgui.h
+++ b/imgui.h
@@ -129,6 +129,7 @@ Index of this file:
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wpragmas"          // warning: unknown option after '#pragma GCC diagnostic' kind
 #pragma GCC diagnostic ignored "-Wclass-memaccess"  // [__GNUC__ >= 8] warning: 'memset/memcpy' clearing/writing an object of type 'xxxx' with no trivial copy-assignment; use assignment or value-initialization instead
+#pragma GCC diagnostic ignored "-Wold-style-cast"
 #endif

 //-----------------------------------------------------------------------------
